### ISSUE TITLE
Unbreakable tools setting causes held item to instantly break if armour hits 0 durability, and armour goes to negative durability

### ISSUE THREAD

**Issue by Hexicube** [SigmundGranaas/forgero#1118]
When unbreakable tools is enabled armour can end up with a negative durability, taking any damage that would cause armour to break instead causes the currently held item to instantly go to 0 durability and lets the armour persist.

This only appears to care about held items, as soon as I switch away from a Forgero item the armour breaks.

**Comment by Hexicube** [comment]
Mod list:

```
Forge Config API Port [8.0.1]
Retraining [1.3.1]
Xaero's Minimap [24.6.1]
Shulker Box Tooltip [4.0.4+1.20.1]
Clumps [12.0.0.4]
wthit [8.15.3]
Explorer's Compass [1.20.1-2.2.3-fabric]
Faux-Custom-Entity-Data [6.0.1]
Resourcefulconfig [2.1.2]
Architectury [9.2.14]
ToolStats [16.0.9]
Waystones [14.1.6]
Lithium [0.11.2]
ModernFix [5.19.5+mc1.20.1]
Extreme sound muffler [3.48]
Patchouli [1.20.1-84-FABRIC]
Default Options [18.0.1]
Enhanced Block Entities [0.9+1.20]
Entity Model Features [2.2.6]
Roughly Enough Professions [2.0.2]
Smelting Sand In Blast a Furnace [1.20.1-11-fabric]
Model Gap Fix [1.15]
Continuity [3.0.0-beta.5+1.20.1]
AppleSkin [2.5.1+mc1.20]
No Hostiles Around Campfire [7.1]
FallingTree [4.3.4]
Convenient mobGriefing [2.1.0]
ImmediatelyFast [1.3.2+1.20.4]
SIHYWTCAMD [1.10.5+1.20.1]
Smelting Cobblestone In Blast a Furnace [Fabric/Forge] [1-all]
Botarium [2.3.4]
Better Mount HUD [1.2.2]
No Chat Reports [1.20.1-v2.2.2]
Cloth Config v11 [11.1.136]
FerriteCore [6.0.1]
More Culling [1.20.4-0.24.0]
Furnace Recycle [2.5]
Enhanced Attack Indicator [1.0.4+1.20]
Larger Ore Veins: Deluxe [1.2.2]
Mod Menu [7.2.2]
Fabric Language Kotlin [1.12.3+kotlin.2.0.21]
Highlight [2.0.1]
Mouse Tweaks [2.26]
Balm [7.3.9]
Xaero's World Map [1.39.0]
Resourceful Lib [2.1.29]
Terralith [2.5.4]
Raw Ores Processing [1.6]
Entity Texture Features [6.2.7]
Immersive Thunder [1.20.1+1.2.2]
Better Advancements [0.4.2.10]
Forgero [0.12.8+1.20.1]
Roughly Enough Items [12.1.785]
Bad Packets [0.4.3]
fakerlib [0.1.4]
EnchantmentDescriptions [17.1.19]
Animatica [0.6.1+1.20.4]
LazyDFU [0.1.3]
Collective [7.87]
ToolTip Fix [1.1.1-1.20]
Right Click Harvest [3.2.3+1.19.x-1.20.1-fabric]
Zoomify [2.14.2+1.20.1]
Tax Free Levels [1.4.1]
Dynamic FPS [3.7.6]
Yeetus Experimentus [2.3.0-build.4+mc1.20.1]
EntityCulling [1.7.1]
Bookshelf [20.2.13]
AttributeFix [21.0.4]
Cobweb [1.0.0]
Accelerated Decay [3.0.1+mc1.20.1]
Zenith [1.2.4-1.20.1]
OptiGUI [2.3.0-beta.6+1.20]
Roughly Enough Resources [2.9.0]
Tectonic [2.4.1b]
YetAnotherConfigLib [3.6.1+1.20.1-fabric]
Fabric API [0.92.2+1.20.1]
Village Spawn Point [4.4]
BetterThanMending [1.3.0]
BitsAndChisels [2.7.3]
Puzzles Lib [8.1.25]
Lootr [0.7.33.81]
JamLib [0.6.1+1.20.x]
Nature's Compass [1.20.1-2.2.3-fabric]
Harvest with ease [9.3.0]
Borderless Mining [1.1.8+1.20.1]
Shifting Wares [2.1.2+1.20.2]
Iris [1.7.5+mc1.20.1]
Sodium [0.5.11+mc1.20.1]
Indium [1.0.34+mc1.20.1]
Distant Horizons [2.2.1-a]
Zenith Attributes [0.2.9]
Ad Astra [1.15.19]
CraftTweaker [14.0.44]
Effortless Structure [3.1.0]
```

**Comment by SigmundGranaas** [comment]
Thanks for the issue! I'll try to reproduce it and see if I can find the cause :+1:
